<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>800</width>
    <height>600</height>
   </rect>
  </property>
  <property name="sizePolicy">
   <sizepolicy hsizetype="MinimumExpanding" vsizetype="MinimumExpanding">
    <horstretch>0</horstretch>
    <verstretch>0</verstretch>
   </sizepolicy>
  </property>
  <property name="minimumSize">
   <size>
    <width>800</width>
    <height>600</height>
   </size>
  </property>
  <property name="windowTitle">
   <string>Herborist</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <property name="sizePolicy">
    <sizepolicy hsizetype="Minimum" vsizetype="Minimum">
     <horstretch>0</horstretch>
     <verstretch>0</verstretch>
    </sizepolicy>
   </property>
   <layout class="QHBoxLayout" name="horizontalLayout_3">
    <item>
     <layout class="QHBoxLayout" name="horizontalLayout_2">
      <item>
       <layout class="QVBoxLayout" name="verticalLayout">
        <item>
         <widget class="QLabel" name="label">
          <property name="font">
           <font>
            <weight>75</weight>
            <bold>true</bold>
           </font>
          </property>
          <property name="text">
           <string>Usb2serial ports:</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QComboBox" name="port_box"/>
        </item>
        <item>
         <widget class="QLabel" name="label_2">
          <property name="font">
           <font>
            <weight>75</weight>
            <bold>true</bold>
           </font>
          </property>
          <property name="text">
           <string>Baudrates:</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QListWidget" name="baudrate_list">
          <property name="alternatingRowColors">
           <bool>false</bool>
          </property>
          <property name="isWrapping" stdset="0">
           <bool>false</bool>
          </property>
          <property name="uniformItemSizes">
           <bool>false</bool>
          </property>
          <property name="selectionRectVisible">
           <bool>false</bool>
          </property>
          <item>
           <property name="text">
            <string>1000000</string>
           </property>
           <property name="checkState">
            <enum>Checked</enum>
           </property>
          </item>
          <item>
           <property name="text">
            <string>500000</string>
           </property>
           <property name="checkState">
            <enum>Checked</enum>
           </property>
          </item>
          <item>
           <property name="text">
            <string>400000</string>
           </property>
           <property name="checkState">
            <enum>Checked</enum>
           </property>
          </item>
          <item>
           <property name="text">
            <string>250000</string>
           </property>
           <property name="checkState">
            <enum>Checked</enum>
           </property>
          </item>
          <item>
           <property name="text">
            <string>200000</string>
           </property>
           <property name="checkState">
            <enum>Checked</enum>
           </property>
          </item>
          <item>
           <property name="text">
            <string>117647</string>
           </property>
           <property name="checkState">
            <enum>Checked</enum>
           </property>
          </item>
          <item>
           <property name="text">
            <string>57600</string>
           </property>
           <property name="checkState">
            <enum>Checked</enum>
           </property>
          </item>
          <item>
           <property name="text">
            <string>19230</string>
           </property>
           <property name="checkState">
            <enum>Checked</enum>
           </property>
          </item>
          <item>
           <property name="text">
            <string>9615</string>
           </property>
           <property name="checkState">
            <enum>Checked</enum>
           </property>
          </item>
         </widget>
        </item>
        <item>
         <widget class="Line" name="line_2">
          <property name="orientation">
           <enum>Qt::Horizontal</enum>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QLabel" name="label_3">
          <property name="text">
           <string>ID ranges to search:</string>
          </property>
         </widget>
        </item>
        <item>
         <layout class="QHBoxLayout" name="horizontalLayout">
          <item>
           <widget class="QLabel" name="label_4">
            <property name="text">
             <string>Minimum</string>
            </property>
            <property name="alignment">
             <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QSpinBox" name="min_id">
            <property name="maximum">
             <number>254</number>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QLabel" name="label_5">
            <property name="text">
             <string>Maximum</string>
            </property>
            <property name="alignment">
             <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QSpinBox" name="max_id">
            <property name="cursor">
             <cursorShape>ArrowCursor</cursorShape>
            </property>
            <property name="maximum">
             <number>253</number>
            </property>
            <property name="value">
             <number>253</number>
            </property>
           </widget>
          </item>
         </layout>
        </item>
        <item>
         <widget class="Line" name="line">
          <property name="orientation">
           <enum>Qt::Horizontal</enum>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QPushButton" name="scan_button">
          <property name="enabled">
           <bool>false</bool>
          </property>
          <property name="sizePolicy">
           <sizepolicy hsizetype="Minimum" vsizetype="Fixed">
            <horstretch>0</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
          <property name="text">
           <string>Scan</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QProgressBar" name="scan_progress">
          <property name="value">
           <number>0</number>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QPushButton" name="abort_button">
          <property name="enabled">
           <bool>false</bool>
          </property>
          <property name="focusPolicy">
           <enum>Qt::TabFocus</enum>
          </property>
          <property name="text">
           <string>Abort</string>
          </property>
         </widget>
        </item>
       </layout>
      </item>
      <item>
       <widget class="QTreeWidget" name="motor_tree">
        <property name="enabled">
         <bool>false</bool>
        </property>
        <property name="acceptDrops">
         <bool>false</bool>
        </property>
        <property name="horizontalScrollBarPolicy">
         <enum>Qt::ScrollBarAlwaysOff</enum>
        </property>
        <property name="autoScroll">
         <bool>true</bool>
        </property>
        <property name="showDropIndicator" stdset="0">
         <bool>false</bool>
        </property>
        <property name="dragEnabled">
         <bool>false</bool>
        </property>
        <property name="dragDropMode">
         <enum>QAbstractItemView::NoDragDrop</enum>
        </property>
        <property name="selectionMode">
         <enum>QAbstractItemView::ExtendedSelection</enum>
        </property>
        <column>
         <property name="text">
          <string>baudrate</string>
         </property>
        </column>
        <column>
         <property name="text">
          <string>id</string>
         </property>
        </column>
        <column>
         <property name="text">
          <string>model</string>
         </property>
        </column>
       </widget>
      </item>
      <item>
       <layout class="QGridLayout" name="motor_layout" columnstretch="0,0,0">
        <item row="7" column="1">
         <widget class="QLCDNumber" name="torque_max_lcd"/>
        </item>
        <item row="11" column="1">
         <widget class="QLCDNumber" name="lower_limit_lcd"/>
        </item>
        <item row="11" column="2">
         <widget class="QLCDNumber" name="upper_limit_lcd"/>
        </item>
        <item row="5" column="0">
         <widget class="QLabel" name="label_11">
          <property name="text">
           <string>Firmware</string>
          </property>
         </widget>
        </item>
        <item row="5" column="1" colspan="2">
         <widget class="QLabel" name="firmware_label">
          <property name="text">
           <string/>
          </property>
          <property name="alignment">
           <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
          </property>
         </widget>
        </item>
        <item row="0" column="0" colspan="3">
         <widget class="QLabel" name="label_6">
          <property name="font">
           <font>
            <weight>75</weight>
            <bold>true</bold>
           </font>
          </property>
          <property name="text">
           <string>Motor Configuration:</string>
          </property>
          <property name="alignment">
           <set>Qt::AlignCenter</set>
          </property>
         </widget>
        </item>
        <item row="14" column="1">
         <widget class="QLabel" name="label_17">
          <property name="text">
           <string>Present</string>
          </property>
          <property name="alignment">
           <set>Qt::AlignCenter</set>
          </property>
         </widget>
        </item>
        <item row="17" column="0" colspan="3">
         <widget class="QCheckBox" name="torque_checkbox">
          <property name="text">
           <string>Torque</string>
          </property>
         </widget>
        </item>
        <item row="3" column="1">
         <widget class="QLCDNumber" name="rdt_lcd"/>
        </item>
        <item row="10" column="0" rowspan="2">
         <widget class="QLabel" name="label_15">
          <property name="sizePolicy">
           <sizepolicy hsizetype="Minimum" vsizetype="Minimum">
            <horstretch>0</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
          <property name="minimumSize">
           <size>
            <width>0</width>
            <height>0</height>
           </size>
          </property>
          <property name="text">
           <string>Angle limits</string>
          </property>
         </widget>
        </item>
        <item row="13" column="0" colspan="3">
         <widget class="Line" name="line_4">
          <property name="lineWidth">
           <number>1</number>
          </property>
          <property name="orientation">
           <enum>Qt::Horizontal</enum>
          </property>
         </widget>
        </item>
        <item row="14" column="0" rowspan="3">
         <widget class="QLabel" name="label_16">
          <property name="sizePolicy">
           <sizepolicy hsizetype="Minimum" vsizetype="Minimum">
            <horstretch>0</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
          <property name="text">
           <string>Position</string>
          </property>
         </widget>
        </item>
        <item row="15" column="1">
         <widget class="QDial" name="present_position_dial">
          <property name="minimum">
           <number>-180</number>
          </property>
          <property name="maximum">
           <number>180</number>
          </property>
          <property name="singleStep">
           <number>18</number>
          </property>
          <property name="pageStep">
           <number>5</number>
          </property>
          <property name="notchesVisible">
           <bool>true</bool>
          </property>
         </widget>
        </item>
        <item row="14" column="2">
         <widget class="QLabel" name="label_18">
          <property name="text">
           <string>Goal</string>
          </property>
          <property name="alignment">
           <set>Qt::AlignCenter</set>
          </property>
         </widget>
        </item>
        <item row="10" column="1">
         <widget class="QDial" name="lower_limit_dial">
          <property name="sizePolicy">
           <sizepolicy hsizetype="Minimum" vsizetype="Minimum">
            <horstretch>0</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
          <property name="minimum">
           <number>-180</number>
          </property>
          <property name="maximum">
           <number>180</number>
          </property>
          <property name="singleStep">
           <number>18</number>
          </property>
          <property name="pageStep">
           <number>5</number>
          </property>
          <property name="notchesVisible">
           <bool>true</bool>
          </property>
         </widget>
        </item>
        <item row="16" column="1">
         <widget class="QLCDNumber" name="present_position_lcd"/>
        </item>
        <item row="3" column="0">
         <widget class="QLabel" name="label_9">
          <property name="text">
           <string>Return delay time</string>
          </property>
         </widget>
        </item>
        <item row="2" column="0">
         <widget class="QLabel" name="label_8">
          <property name="text">
           <string>Baudrate</string>
          </property>
         </widget>
        </item>
        <item row="12" column="0" colspan="3">
         <widget class="QPushButton" name="update_eeprom_button">
          <property name="text">
           <string>Update EEPROM</string>
          </property>
         </widget>
        </item>
        <item row="7" column="0">
         <widget class="QLabel" name="label_14">
          <property name="text">
           <string>Torque max</string>
          </property>
         </widget>
        </item>
        <item row="2" column="1" colspan="2">
         <widget class="QComboBox" name="baud_combobox">
          <item>
           <property name="text">
            <string>1000000</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>500000</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>400000</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>250000</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>200000</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>117647</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>57600</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>19230</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>9615</string>
           </property>
          </item>
         </widget>
        </item>
        <item row="1" column="1" colspan="2">
         <widget class="QSpinBox" name="id_spinbox">
          <property name="minimum">
           <number>0</number>
          </property>
          <property name="maximum">
           <number>253</number>
          </property>
         </widget>
        </item>
        <item row="16" column="2">
         <widget class="QLCDNumber" name="goal_position_lcd"/>
        </item>
        <item row="1" column="0">
         <widget class="QLabel" name="label_7">
          <property name="text">
           <string>ID</string>
          </property>
         </widget>
        </item>
        <item row="4" column="0">
         <widget class="QLabel" name="label_10">
          <property name="text">
           <string>Status return level</string>
          </property>
         </widget>
        </item>
        <item row="10" column="2">
         <widget class="QDial" name="upper_limit_dial">
          <property name="sizePolicy">
           <sizepolicy hsizetype="Minimum" vsizetype="Minimum">
            <horstretch>0</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
          <property name="minimum">
           <number>-180</number>
          </property>
          <property name="maximum">
           <number>180</number>
          </property>
          <property name="singleStep">
           <number>18</number>
          </property>
          <property name="pageStep">
           <number>5</number>
          </property>
          <property name="notchesVisible">
           <bool>true</bool>
          </property>
         </widget>
        </item>
        <item row="15" column="2">
         <widget class="QDial" name="goal_position_dial">
          <property name="minimum">
           <number>-180</number>
          </property>
          <property name="maximum">
           <number>180</number>
          </property>
          <property name="singleStep">
           <number>18</number>
          </property>
          <property name="pageStep">
           <number>5</number>
          </property>
          <property name="notchesVisible">
           <bool>true</bool>
          </property>
         </widget>
        </item>
        <item row="7" column="2">
         <widget class="QSlider" name="torque_max_slider">
          <property name="maximum">
           <number>100</number>
          </property>
          <property name="orientation">
           <enum>Qt::Horizontal</enum>
          </property>
          <property name="tickPosition">
           <enum>QSlider::TicksBelow</enum>
          </property>
         </widget>
        </item>
        <item row="4" column="1" colspan="2">
         <widget class="QComboBox" name="srl_combobox">
          <item>
           <property name="text">
            <string>Never</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>Read</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>Always</string>
           </property>
          </item>
         </widget>
        </item>
        <item row="3" column="2">
         <widget class="QSlider" name="rdt_slider">
          <property name="maximum">
           <number>500</number>
          </property>
          <property name="orientation">
           <enum>Qt::Horizontal</enum>
          </property>
          <property name="tickPosition">
           <enum>QSlider::TicksBelow</enum>
          </property>
          <property name="tickInterval">
           <number>50</number>
          </property>
         </widget>
        </item>
       </layout>
      </item>
     </layout>
    </item>
   </layout>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
 </widget>
 <resources/>
 <connections/>
</ui>
